hoomd/Variant.h

```diff
@@ -50,27 +50,330 @@ class PYBIND11_EXPORT VariantConstant : public Variant
             {
             }
 
+        /// Return the value.
         Scalar operator()(uint64_t timestep)
             {
             return m_value;
             }
 
-        /// Set the value
+        /// Set the value.
         void setValue(Scalar value)
             {
             m_value = value;
             }
 
-        /// Get the value
+        /// Get the value.
         Scalar getValue()
             {
             return m_value;
             }
 
     protected:
-        /// The value
+        /// The value.
         Scalar m_value;
     };
 
+/** Ramp variant.
+
+    Variant that ramps linearly from A to B over a given number of steps.
+*/
+class PYBIND11_EXPORT VariantRamp : public Variant
+    {
+    public:
+
+        /** Construct a VariantRamp.
+
+            @param A The starting value.
+            @param B The ending value.
+            @param t_start The starting time step.
+            @param t_ramp The length of the ramp.
+        */
+        VariantRamp(Scalar A, Scalar B, uint64_t t_start, uint64_t t_ramp)
+            {
+            setA(A);
+            setB(B);
+            setTStart(t_start);
+            setTRamp(t_ramp);
+            }
+
+        /// Evaluate the ramp.
+        Scalar operator()(uint64_t timestep)
+            {
+            if (timestep < m_t_start)
+                {
+                return m_A;
+                }
+            else if (timestep < m_t_start + m_t_ramp)
+                {
+                // interpolate between A and B
+                uint64_t v = timestep - m_t_start;
```

Comment:
Since we cast `v` to a `double`, is there a reason that this line is here and we don't use
````C++
auto s = double(timestep - m_t_start) / double(m_t_ramp);
````

Reply:
Thanks, this is a good simplification.